usr/src/uts/common/fs/zfs/arc.c

```diff
@@ -6882,10 +6882,10 @@ l2arc_write_buffers(spa_t *spa, l2arc_dev_t *dev, uint64_t target_sz)
 
 	ASSERT3U(write_asize, <=, target_sz);
 	ARCSTAT_BUMP(arcstat_l2_writes_sent);
-	ARCSTAT_INCR(arcstat_l2_write_bytes, write_asize);
+	ARCSTAT_INCR(arcstat_l2_write_bytes, write_psize);
 	ARCSTAT_INCR(arcstat_l2_size, write_sz);
```

Comment:
my suggestion was to change the arcstat to have more meaning (i.e. arcstat_l2_lsize) but keep the kstat name the same (i.e. "l2_size"). This shouldn't break any external consumers and will allow developers to know the true meaning of this counter.